parcels/particlefile.py

```diff
@@ -162,17 +163,20 @@ def write(self, pset, time, sync=True, deleted_only=False):
                     if p.dt*p.time <= p.dt*time:  # don't write particles if they haven't started yet
                         i = p.fileid
                         self.id[i, self.idx[i]] = p.id
-                        self.time[i, self.idx[i]] = time
+                        self.time[i, self.idx[i]] = p.time
                         self.lat[i, self.idx[i]] = p.lat
                         self.lon[i, self.idx[i]] = p.lon
                         self.z[i, self.idx[i]] = p.depth
                         for var in self.user_vars:
                             getattr(self, var)[i, self.idx[i]] = getattr(p, var)
+                        if p.state != ErrorCode.Delete:
+                            assert np.allclose(p.time, time), \
```

Comment:
Shouldn't this be a warning? Do we really want Parcels to crash if this happens?